### DISCUSSION TITLE
responsive image size, performance for mobile devices

### DISCUSSION THREAD

**Discussion by adamus007p** [PrestaShop/PrestaShop#37686] [Ideas]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hi there, I was new classic theme for 8.2.
I would like to share with my solution which it give few image version.

`<picture><source srcset=" https://www.domain8.com/123-home_default/image.webp 640w, https://www.domain8.com/123-medium_default/image.webp 916w" type="image/webp" sizes="(min-width: 800px) 1030px, 100vw"><source srcset=" https://www.domain8.com/123-home_default/image.jpg 640w, https://www.domain8.com/123-medium_default/image.jpg 916w" type="image/jpeg" sizes="(min-width: 1536px) 1030px, 100vw"><img class="js-qv-product-cover img-fluid" src="https://www.domain8.com/123-medium_default/image.jpg" alt="Product name" title="Product name"></picture>`

instead
`<picture>
                    <source srcset="https://www.domain.com/123-large_default/product.webp" type="image/webp">          <img class="js-qv-product-cover img-fluid" src="https://www.domain.com/123-large_default/product.jpg" alt="Product name" title="Product name" loading="lazy" width="800" height="800">
        </picture>`

Now in classic it gives a big image for mobile devices which is a waste of resources and make load time slower..

@Hlavtox @kpodemski  you probably will be interested of this

### Steps to reproduce

N/A

### PrestaShop version(s) where the bug happened

8.2

**Comment by Hlavtox** [comment]
We already did this in hummingbird - https://github.com/PrestaShop/hummingbird/blob/develop/templates/catalog/_partials/product-cover-thumbnails.tpl :-)

But it's doesn't work like it should... 

If you have an image 800px wide and sizes 320x320, 720x720 and 1440x1440.

You get image with huge white borders for 1440x1440 thumbnail. Which results in bad display in browsers that decide to use it.

**Comment by adamus007p** [comment]
@Hlavtox from where that borders comes? Is it related to image regeneration?

**Comment by Hlavtox** [comment]
@adamus007p It takes the original image:
- If the thumbnail is smaller, it shrinks down the original image.
- If the thumbnail is bigger, it adds white borders to the original image.

**Comment by adamus007p** [comment]
@Hlavtox Is there any demo that I can look at it?

Is it done only i develop or in master too?
